src/frontend/Elaborator.ml

```diff
@@ -190,11 +190,14 @@ and chk_tm_in_tele (args : CS.cell list) (con : CS.con) : T.Chk.tac =
   let rec loop args =
     match args with
     | [] -> chk_tm con
-    | CS.Cell {name; tp} :: args ->
-      T.Chk.update_span tp.info @@
-      Tactics.intro_implicit_connectives @@
-      R.Pi.intro ~ident:name @@ fun _ ->
-      loop args
+    | CS.Cell {names; tp} :: args ->
+      List.fold_right
+        (fun name body ->
+           T.Chk.update_span tp.info @@
+           Tactics.intro_implicit_connectives @@
+           R.Pi.intro ~ident:name @@ fun _ -> body)
+        names
+        (loop args)
   in
   loop args
```

Comment:
This function is ugly because I don't want to actually think what the function is doing. Instead, a mechanical translation was done.